### PR TITLE
Update webview.cpp

### PR DESCRIPTION
**Pull request by Ritvyk**
Comments are added to make the code easy to understand.

### CODE REVIEW

**Comment by kelson42** [comment]
@Ritvyk thx for your commit, this will be reviewed in a few days by a developer. Please try to take a "good first issue" if you want to help Kiwix Desktop https://github.com/kiwix/kiwix-desktop/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22

File: src/webview.cpp [hunk 1]
```diff
@@ -1,53 +1,53 @@
...
 #include <iostream>
 #include "kiwixapp.h"
 #include "webpage.h"
 
 WebView::WebView(QWidget *parent)
-    : QWebEngineView(parent)
+    :QWebEngineView(parent)
 {
-    setPage(new WebPage(this));
+    setPage(new WebPage(this));   //to show the ZIM file as a webpage
```
Comment (mgautierfr) [inline src/webview.cpp]: No.
We need a custom webpage to be able to open a external browser when the user click on a external link.
To do so, we need to overwrite `bool WebPage::acceptNavigationRequest` to check all request in the webView.
See https://github.com/kiwix/kiwix-desktop/blob/master/src/webpage.cpp

File: src/webview.cpp [hunk 2]
```diff
@@ -1,53 +1,53 @@
...
     auto profile = page()->profile();
     auto app = KiwixApp::instance();
     profile->installUrlSchemeHandler("zim", app->getSchemeHandler());
     QObject::connect(this, &QWebEngineView::urlChanged, this, &WebView::onUrlChanged);
 }
 
-WebView::~WebView()
+WebView::~WebView()                                     //gaussian max destructor
```
Comment (mgautierfr) [inline src/webview.cpp]: What is a "gaussian max destructor" ?

File: src/webview.cpp [hunk 3]
```diff
@@ -1,53 +1,53 @@
...
 }
 
-QWebEngineView* WebView::createWindow(QWebEnginePage::WebWindowType type)
+QWebEngineView* WebView::createWindow(QWebEnginePage::WebWindowType type)       //Function to create webview window
 {
     if ( type==QWebEnginePage::WebBrowserBackgroundTab
       || type==QWebEnginePage::WebBrowserTab )
     {
-        auto tabWidget = KiwixApp::instance()->getTabWidget();
+        auto tabWidget = KiwixApp::instance()->getTabWidget();      //to apply the Tab Widget
```
Comment (mgautierfr) [inline src/webview.cpp]: I don't understand your comment here.
To apply what ?

File: src/webview.cpp [hunk 4]
```diff
@@ -1,53 +1,53 @@
...
+{
     auto zimId = url.host();
     zimId.resize(zimId.length()-4);
     if (m_currentZimId != zimId ) {
-        m_currentZimId = zimId;
+        m_currentZimId = zimId;             //to retrive the correct ZIM files
         emit zimIdChanged(m_currentZimId);
         auto app = KiwixApp::instance();
-        auto reader = app->getLibrary()->getReader(m_currentZimId);
+        auto reader = app->getLibrary()->getReader(m_currentZimId);  //reading the Zim files from the server
```
Comment (mgautierfr) [inline src/webview.cpp]: No, we get it from the library, as the code implies.

File: src/webview.cpp [hunk 5]
```diff
@@ -1,53 +1,53 @@
...
         std::string favicon, _mimetype;
-        reader->getFavicon(favicon, _mimetype);
+        reader->getFavicon(favicon, _mimetype);     //setting the favicon icon
```
Comment (mgautierfr) [inline src/webview.cpp]: We are not setting anything here, we are getting the icon to set from the reader.

File: src/webview.cpp [hunk 6]
```diff
@@ -1,53 +1,53 @@
...
         QPixmap pixmap;
         pixmap.loadFromData((const uchar*)favicon.data(), favicon.size());
-        m_icon = QIcon(pixmap);
+        m_icon = QIcon(pixmap);                 //pasting the favicon icon
```
Comment (mgautierfr) [inline src/webview.cpp]: We are creating an QIcon from the raw data. This icon will be used by the handle of the `iconChanged` signal.

**Comment by kelson42** [comment]
@Ritvyk Any feedback on this PR?

**Comment by kelson42** [comment]
No feedback, I close the PR.

**Pull request closed without merging**